tests/pages/api/multipleProducts.test.ts

```diff
@@ -170,4 +123,75 @@ describe('multipleProduct', () => {
         multipleProduct(req, res);
         expect(writeHeadMock).toBeCalledWith(302, expectedLocation);
     });
+
+    describe('getErrorsForCookie', () => {
+        it('returns an empty array if a product list has no errors', () => {
+            const errors = getErrorsForCookie(multipleProducts);
+            expect(errors).toEqual([]);
+        });
+
+        it('returns an error info array if a product list has errors', () => {
+            const errors = getErrorsForCookie(multipleProductsWithErrors);
+            expect(errors.length).toBe(2);
+        });
+    });
+
+    describe('containsErrors', () => {
+        it('returns true if a product list has errors', () => {
+            const result = containsErrors(multipleProductsWithErrors);
+            expect(result).toBe(true);
+        });
+
+        it('returns false if a product list has no errors', () => {
+            const result = containsErrors(multipleProducts);
+            expect(result).toBe(false);
+        });
+    });
+
+    describe('checkProductDurationsAreValid', () => {
+        it('adds duration errors to a product with invalid durations', () => {
+            const result = checkProductDurationsAreValid(invalidDurationProducts);
+            expect(result[0].productDurationError).toBeUndefined();
+            expect(result[1].productDurationError).toBe('Product duration cannot be zero or a negative number');
+            expect(result[2].productDurationError).toBe('Product duration cannot be zero or a negative number');
+            expect(result[3].productDurationError).toBe('This field cannot be empty');
+            expect(result[4].productDurationError).toBe('Product duration must be a whole, positive number');
+        });
+    });
+
+    describe('checkProductDurationTypesAreValid', () => {
+        it('adds duration errors to a product with invalid durations', () => {
+            const result = checkProductDurationTypesAreValid(invalidDurationTypeProducts);
+            expect(result[0].productDurationUnitsError).toBeUndefined();
+            expect(result[1].productDurationUnitsError).toBe('Choose an option from the dropdown');
+            expect(result[2].productDurationUnitsError).toBeUndefined();
+        });
```

Comment:
This third test appears to test the same thing as that on line 165 (the `valid duration type` and `valid duration` products in the `invalidDurationTypeProducts` mock appear to be identical?)